main.go

```diff
@@ -39,44 +51,42 @@ var secretsGeneration = flag.String(
 	"Secrets Generation (rotation counter)",
 )
 
-var certExpiration = flag.Int(
-	"certExpiration",
-	30*365+7, // just over 30 years
-	"Certificate expiration (in days)",
+var secretsName = flag.String(
+	"secretsName",
+	"",
+	"Secrets Name (version string of the helm chart)",
 )
 
 func main() {
 	flag.Parse()
 
-	if flag.NArg() != 1 {
-		fmt.Fprintf(flag.CommandLine.Output(), "Usage: %s [OPTIONS] ROLE-MANIFEST\n", os.Args[0])
-		flag.PrintDefaults()
+	if *installMode != "install" && *installMode != "upgrade" {
+		fmt.Fprintf(flag.CommandLine.Output(), "Invalid -mode: `%s`, must be either `install` or `upgrade`.\n", *installMode)
+		flag.Usage()
 		os.Exit(1)
 	}
 
 	sg := secrets.SecretGenerator{
-		Domain:              *domain,
-		Namespace:           *namespace,
-		ServiceDomainSuffix: *serviceDomainSuffix,
-		SecretsName:         *secretsName,
-		SecretsGeneration:   *secretsGeneration,
-		CertExpiration:      *certExpiration,
+		CertExpiration:    *certExpiration,
+		ClusterDomain:     *clusterDomain,
+		Domain:            *domain,
+		IsInstall:         (*installMode == "install"),
+		Namespace:         *namespace,
+		SecretsGeneration: *secretsGeneration,
+		SecretsName:       *secretsName,
 	}
 	if sg.Domain == "" {
 		log.Fatal("-domain is not set")
 	}
 	if sg.Namespace == "" {
 		log.Fatal("-namespace is not set")
 	}
-	if sg.ServiceDomainSuffix == "" {
-		log.Fatal("-serviceDomainSuffix is not set")
```

Comment:
No checks for `--clusterDomain` to be set? (As in, `scf-secret-generator --clusterDomain="" …`)

Reply:
Because it defaults to `cluster.local`